Sources/X509/CSR/CSRAttribute.swift

```diff
@@ -0,0 +1,122 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftCertificates open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftCertificates project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftCertificates project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftASN1
+
+extension CertificateSigningRequest {
+    /// A general-purpose representation of a ``CertificateSigningRequest`` attribute.
+    ///
+    /// ``CertificateSigningRequest``s can have a number of attributes applied to them. These attributes are non-binding, but can be
+    /// used to encode things like extensions that should be applied to the generated certificate, or challenge material that can
+    /// be used to enable later revocation.
+    ///
+    /// ``Attribute``s are a general representation, keyed by an object identifier and storing a `SET` of arbitrary values.
+    /// This `SET` cannot be empty. Individual ``Attribute``s can be parsed into more specific values as-needed.
+    public struct Attribute {
+        /// The identifier for this attribute type.
+        ///
+        /// Common values are stored in `ASN1ObjectIdentifier.X509ExtensionID`.
+        public var oid: ASN1ObjectIdentifier
+
+        /// The encoded bytes of the values of this attribute.
+        ///
+        /// This value should be decoded based on the value of ``oid``.
+        public var values: [ASN1Any]
+
+        /// Construct a new attribute from its constituent parts.
+        ///
+        /// - Parameters:
+        ///   - oid: The identifier for this extension type.
+        ///   - values: The value of this attribute, erased to `ASN1Any`
+        @inlinable
+        public init(oid: ASN1ObjectIdentifier, values: [ASN1Any]) {
+            self.oid = oid
+            self.values = values
+        }
+    }
+}
+
+extension CertificateSigningRequest.Attribute: Hashable {
+    @inlinable
+    public static func ==(lhs: CertificateSigningRequest.Attribute, rhs: CertificateSigningRequest.Attribute) -> Bool {
+        if lhs.oid != rhs.oid { return false }
+
+        for element in lhs.values {
```

Comment:
`lhs.values.count == rhs.values.count` must be true as well or they will be equal even if `rhs` has additional elements.
Can we add for it as well?

Same needs to be done for `CertificateSigningRequest.Attributes`

Reply:
Oh good catch!